plugins/ru.runa.gpd.form.ftl/src/ru/runa/gpd/formeditor/ftl/validation/DefaultComponentValidator.java

```diff
@@ -1,19 +1,27 @@
 package ru.runa.gpd.formeditor.ftl.validation;
 
+import java.text.MessageFormat;
 import java.util.List;
 
+import ru.runa.gpd.Localization;
 import ru.runa.gpd.formeditor.ftl.Component;
 import ru.runa.gpd.formeditor.ftl.ComponentParameter;
 import ru.runa.gpd.lang.ValidationError;
 import ru.runa.gpd.lang.model.FormNode;
 
+import com.google.common.base.Objects;
 import com.google.common.collect.Lists;
 
 public class DefaultComponentValidator implements IComponentValidator {
 
     @Override
-    public List<ValidationError> validate(FormNode formNode, Component component) {
+    public List<ValidationError> validate(FormNode formNode, Component component, byte[] formData) {
         List<ValidationError> list = Lists.newArrayList();
+        
+        String componentParameters = new String(formData);
```

Comment:
What is this?

Reply:
We need to check whether the component model has same parameters as written in form. In this line we get component string as in form.

Reply:
But formData represents many components and html... not only selected component parameters...